### ISSUE TITLE
Conditional profiles will not work with d3d11va set.

### ISSUE THREAD

**Issue by JySzE** [mpv-player/mpv#14042]
### Important Information

Provide following Information:
- mpv-x86_64-v3-20240421-git-b364e4a & v0.38.0
- Windows 11
- https://sourceforge.net/projects/mpv-player-windows/files/64bit-v3/
- If known which version of mpv introduced the problem: I do not.
- GPU model, driver and version: Nvidia RTX 3070* 552.22 

### Reproduction steps

Use provided [mpv.conf ](https://pastebin.com/4JmY2nCs) for testing or simply create your own conditional profile and set hwdec to d3d11va: 

Conditional profiles will not activate with `d3d11va` set but will work with `d3dva11-copy` set.

### Expected behavior

Conditional profiles activating. 

### Actual behavior

Conditional profiles not activating. 

### Log file

[output.txt](https://github.com/mpv-player/mpv/files/15190922/output.txt)

### Sample files

[mpv.conf ](https://pastebin.com/4JmY2nCs)

**Comment by JySzE** [comment]
> hwdec generally doesn't work great in auto profiles, even copyback suffers from the exact same issue.
> 
> The issue is that the autoprofile is applied after the video is already initialized, so when you have an autoprofile like
> 
> ```
> [2160p]
> profile-cond=p.height == 2160
> profile-restore=copy
> hwdec=vulkan-copy
> ```
> 
> What happens is that the mpv inits with software decoding, then it applies hwdec at runtime. This ends up breaking the first few seconds of the video, which is fairly normal if you switch to hwdec during runtime. The problem is even more apparent if you do a reverse profile like
> 
> ```
> hwdec=vulkan-copy
> 
> [no-2160p]
> profile-cond=p.height < 2160
> profile-restore=copy
> hwdec=no
> ```
> 
> ... where you can pretty consistently trigger an A/V desync at the start of files
> 
> tl;dr, don't bother putting hwdec in autoprofiles.

Thanks for the reply but if you look at my mpv.conf provided im not switching hwdec in the auto profiles. 

Im setting `hwdec=d3d11va` as the global hwdec. The auto profiles do **not** change what has already been set for `hwdec`

And the auto profiles works fine when set to `hwdec=d3d11va-copy` globally but not when `hwdec=d3d11va` is set.

**Comment by Jules-A** [comment]
Changing shaders in auto-profile with conditional shaders no longer works with d3d11va. Though maybe it's a separate issue since the error is saying it can't find the shader files?

**Comment by JySzE** [comment]
> Everything works as expected, if you use native hwdec, you should use `hw-pixelformat` to get the real format.

Thank you this fixed the issue.
